Content.Shared/Clothing/SharedMagbootsSystem.cs

```diff
@@ -147,6 +147,17 @@ private void OnIsWeightless(Entity<MagbootsComponent> ent, ref InventoryRelayedE
         if (!ent.Comp.On)
             return;
 
+        // do not cancel weightlessness if the person is in space.
+        if (ent.Comp.RequiresGrid)
+        {
+            var xform = Transform(args.Args.Entity);
+            if (TryComp<GravityComponent>(xform.GridUid, out var gravity) && gravity.Enabled ||
+                TryComp<GravityComponent>(xform.MapUid, out var mapGravity) && mapGravity.Enabled)
+            {
+                return;
+            }
+        }
+
         args.Args.IsWeightless = false;
```

Comment:
This is called every single tick for mob movement so should really be using entity queries.

Reply:
IsWeightless should probably get cached in the same way as can-move or movement speed modifiers.